core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3144,15 +3144,29 @@ public void activeSyncMetadata(AlluxioURI path, Collection<AlluxioURI> changedFi
 
   @Override
   public boolean recordActiveSyncTxid(long txId, long mountId) {
-    try (RpcContext rpcContext = createRpcContext()) {
+    MountInfo mountInfo = mMountTable.getMountInfo(mountId);
+    AlluxioURI mountPath = null;
+    if (mountInfo != null) {
+      mountPath = mountInfo.getAlluxioUri();
+    } else {
+      return false;
+    }
 
+    LockingScheme lockingScheme = new LockingScheme(mountPath, LockPattern.READ, false);
+    try (RpcContext rpcContext = createRpcContext();
+         LockedInodePath inodePath = mInodeTree
+             .lockInodePath(lockingScheme.getPath(), lockingScheme.getMode())) {
```

Comment:
what if the path doesn't exist, e.g. due to a delete or unmount?

Reply:
if the path does not exist, it will just throw a `InvalidPathException`, which would be caught and not write to the journal. I think this is what we want since the write to the journal is entirely optional as a performance optimization.

Reply:
For that I think you need to use `lockFullInodePath`